### DISCUSSION TITLE
asr-summarization-pipeline-simplicity/

### DISCUSSION THREAD

**Discussion by giscus** [amir-pourmand/hugo-papermod-farsi-template#34] [Announcements]
# asr-summarization-pipeline-simplicity/

همون‌طور که می‌دونید، الان مدل‌های زبانی بزرگ خیلی قوی شدند و باهاشون میشه کارهای جالبی انجام داد. یکی از این کاربردها که امروز می‌خوام بهش بپردازم، خلاصه‌سازی متن هست.
ماه پیش فایل‌های استاد ملکیان راجع به سادگی و ساده‌زیستی رو گوش دادم. با خودم گفتم خوبه که اول بیام صوت استاد رو با مدل‌های ASR به متن تبدیل کنم و بعد از اون با استفاده از مدل‌های زبانی بزرگ خلاصه‌اش کنم.

https://aprd.ir/asr-summarization-pipeline-simplicity/

**Comment by pourmandmaryam** [comment]
سلام
 تمامی ابعاد سادگی و ساده زیستی برام قابل درک و مفیده جز،اینکه وقتی ناراحتی خودتو هم ناراحت نشون بدی؛چه اهمیتی برای دیگران داره؟
یا چه فایده ای داره ناراحتیمو نشون بدم وقتی کسی نمیتونه حال منو خوب کنه؟

**Reply by pourmand1376** [reply]
سلام مریم جان. 

اینطور که من فهمیدم بحث اینه که کلاً آدم براش مهم نباشه که بقیه چه فکری می‌کنند و قاعدتاً آدمی که براش مهم نباشه، دلیلی نداره که خوشحالی و ناراحتی‌اش رو پنهان کنه.

اگر حال داشتی تو دانشگاه فایل‌هاش رو دانلود کن و گوش بده. از خلاصه‌ای که نوشتم خیلی بهتره (و یه جاهایی هم حوصله‌ات سر می‌ره از بس کند می‌ره جلو).  

من موقع گوش دادن این فایل دائم به یکی از دوستانم فکر می‌کردم که همیشه به وضوح می‌گفت که «من یک سری ماسک در مقابل مردم دارم و جلوی تقریباً هیچ کس این ماسک‌ها رو کنار نمی‌گذارم». 

به حدی ماسک داشت که مثلاً امروز فلان غذا رو می‌خورد و به به و چه چه می‌کرد و بعد یک ماه می‌فهمیدی که اصلاً اون غذا رو دوست نداشته و بدش میومده. تهش هم به هیچ کدوم از بچه‌ها تو گروه دوستی‌مون خبر نداد و رفت کانادا. 

و من بهش حق میدم چون جلوی من هم ماسک داشت و هیچ وقت نمی‌فهمیدم الان از این چیزی که داره میگه خوشش میاد به خاطر من میگه یا این که واقعاً خوشش میاد. 

فکر می‌کنم زندگی کردن تو دنیایی که آدم‌ها کلا ماسک نداشته باشند، خیلی قشنگ تر از زندگی تو دنیای ماست. اینطوری آدم زودتر می‌تونه از رفتارهای بقیه فیدبک بگیره و خودش رو اصلاح کنه. 

این بحث‌ها، مخصوصاً ما ایرانی‌ها که بخش بزرگی از زندگی‌مون بر اساس «حرف مردم» پیش میره، مفیده.

**Comment by HashemAlavi** [comment]
سلام امیر جان

منم میخوام با این مدل‌های ASR کار کنم. یکم سرچ کردم برای زبان فارسی به نظر Kaldi و Vosk مدل‌های خوبی هستن. منتها مشکل اینه که من نمیتونم نصبشون کنم و باهاشون کار کنم! میشه یک راهنمایی هم در مورد استفاده از مدل‌های ASR بنویسی بیزحمت؟ و اینکه خودت از کدوم مدل‌ها استفاده می‌کنی و کدومش برای چه کاری و هدفی بهتره؟

**Reply by pourmand1376** [reply]
سلام هاشم جان. 

این مدل‌هایی که به درستی پیدا کردی رو باید یکی بشینه با صدها ساعت ویس آموزش بده و اینطور نیست که همین الان قابل استفاده باشه. 

من خودم برای کاربرد داخل سازمانی یک مدل ASR آموزش دادم و قاعدتاً نمی‌تونم مدل رو جایی عرضه کنم؛ چون از منابع شرکت استفاده کردم. 

این فایل‌ها رو هم برای تست عملکرد بهش دادم و ازش خروجی‌ای که می‌بینی رو گرفتم. 

اگر عمر و فرصتی باشه به زودی با کولب می‌شینم یک مدل فارسی تر و تمیز آموزش میدم و همون رو هم منتشر می‌کنم که نیاز نباشه کسی برای چیز ساده‌ای مثل این پول بده. 

ولی فعلاً تنها گزینه‌هایی که مدل به درد بخور ASR برای فارسی دارند، گزینه‌های پولی هست. منظورم سایت‌هایی مثل [آیپا](https://aipaa.ir/#home) هست که کارشون تخصصی همینه.

پی‌نوشت: الان مجدداً سرچ کردم ظاهراً نسخه فارسی مدل‌هایی که گفتی هم موجوده ولی مطمئن نیستم که کیفیت قابل‌قبولی داشته باشند.

**Comment by najmehabdolinasab** [comment]
سلام جناب برومند طی سرچی که اخیرا در مورد کار تحقیقاتی  ام داشتم متوجه شدم کار پایان نامه ای که شما پرزنت کردید بسیار شبیه به موضوع پایان نامه من هست امکانش هست لطف کنید در جهت پیش برد این مسیر بنده رو راهنمایی بفرمایید متاسفانه صفحه چت خصوصی پیدا نکردم برای پرسیدن ابهاماتم

**Reply by pourmand1376** [reply]
سلام. به ایمیل من پیام بدید صحبت کنیم. ایمیلم رو [اینجا](https://amirpourmand.ir/) بردارید.

**Comment by mhm322** [comment]
سلام جناب پورمند
میخواستم درباره این اسپیس سوال کنم که آیا این مدل کارایی مناسبی داره یا خیر
متاسفانه موفق نشدم فایل خودم رو داخلش تست کنم چون خطا میداد
و در نهایت اینکه چطور میتونم این مدل رو به صورت آفلاین استفاده کنم
ممنونم

https://huggingface.co/spaces/pourmand1376/m3hrdadfi-wav2vec2-large-xlsr-persian-v3

**Comment by pourmand1376** [comment]
سلام. وقت شما بخیر. 
امیدوارم حالتون خیلی خوب باشه.

اگر می‌خواید مدل رو آفلاین استفاده کنید، می‌تونید از داکر استفاده کنید. داخل Huggingface گزینه‌ای به نام Run With Docker هست که پنجره‌ای شبیه زیر رو بهتون نشون میده. 
![image](https://github.com/pourmand1376/hugo-papermod-farsi-template/assets/32064808/44c9f75d-bd53-4cc9-94f9-9a386e1b6b10)

بعد همین رو داخل ترمینال کپی می‌کنید و با داکر جواب می‌گیرید. 

ولی به نظرم مدل‌های اوپن‌سورس فارسی تو زمینه ASR چندان کارآمد نیستند و عملکرد این مدل هم خوب نبود.
